### ISSUE TITLE
CNN Advanced 기법 핵심 정리

### ISSUE THREAD

**Issue by vmtmxmf5** [vmtmxmf5/Python-ML-DNN#74]
### Batch Normalization
- 감마와 베타는 역전파로 학습
- 뮤와 시그마는 ~역전파. (Non-trainable vari. 학습데이터로 이동평균)

약한 **규제 효과**가 있음 -> **일반화 성능** 향상(최적화 + overfitting 방지)

### Inception Module V2 이후
- 1x1 Conv : **Bottle Neck** 구조로 **파라미터를 줄인**다
(padding='same' 이면, H, W는 같고 C_out만 바뀐다)

### ResNet - skip connection
skip connection을 추가하면 F(x)에서 무엇을 학습하는가?
> F(x) = h(x) - x

y는 고정이므로 addition 파트에서 더해주는 x를 제외한 값(=**잔차**)만큼 학습한다

### ResNet - identity mapping
고전 ResNet 모델에서 addition 후 activation 을 해주는 단계를 제거
**Pre-activation**을 활용해서 **skip connection** 부분에 activation을 넣는다.

- 왜?
f(h(x) + F(x)) 구조가 x + F(x)로 단순해지기 때문에! (h(x)는 identity)
**단순하면 일반화 성능 증가**(최적화 + overfitting 방지)
